src/library-authoring/backup-restore/LibraryBackupPage.tsx

```diff
@@ -0,0 +1,211 @@
+import { getConfig } from '@edx/frontend-platform';
+import {
+  Alert,
+  Button,
+  Container,
+} from '@openedx/paragon';
+import {
+  useCallback,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import { Helmet } from 'react-helmet';
+
+import { useIntl } from '@edx/frontend-platform/i18n';
+import { Download, Loop, Newsstand } from '@openedx/paragon/icons';
+import SubHeader from '@src/generic/sub-header/SubHeader';
+import { LibraryBackupStatus } from '@src/library-authoring/backup-restore/data/constants';
+import { useCreateLibraryBackup, useGetLibraryBackupStatus } from '@src/library-authoring/backup-restore/data/hooks';
+import NotFoundAlert from '@src/generic/NotFoundAlert';
+import Header from '@src/header';
+import { useLibraryContext } from '@src/library-authoring/common/context/LibraryContext';
+import { useContentLibrary } from '@src/library-authoring/data/apiHooks';
+import messages from '@src/library-authoring/backup-restore/messages';
+
+export const LibraryBackupPage = () => {
+  const intl = useIntl();
+  const { libraryId } = useLibraryContext();
+  const [taskId, setTaskId] = useState<string>('');
+  const [isMutationInProgress, setIsMutationInProgress] = useState<boolean>(false);
+  const timeoutRef = useRef<NodeJS.Timeout | null>(null);
```

Comment:
```suggestion
  const timeoutRef = useRef<ReturnType<typeof setTimeout> | null>(null);
```
This usually runs in a browser, so `NodeJS.Timeout` isn't really the correct type. Browsers usually use `number` as the timer ID type. This way ^ of writing it will work in both NodeJS and browsers.